### PR TITLE
feat: add restartPolicy to `kube_pod_init_container_info` metric

### PR DESCRIPTION
**Pull request by changhyuni**
**What this PR does / why we need it**:

This PR adds a label to `kube_pod_init_container_info` to track the `restartPolicy` set in the init_container
With sidecar support starting with kubernetes 1.28, `init_container` can be used `restartPolicy`
This metric tells you which pods are using the sidecar feature

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
adds a label that is never updated for a given init container

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by mrueg** [comment]
@changhyuni any plans to revive this one? This could be an interesting label

**Comment by changhyuni** [comment]
> @changhyuni any plans to revive this one? This could be an interesting label
@mrueg I've reopened it. I'll fix the tests that are failing.

File: internal/store/pod.go [hunk 1]
```diff
@@ -677,16 +677,21 @@ func createPodInitContainerInfoFamilyGenerator() generator.FamilyGenerator {
 		"",
 		wrapPodFunc(func(p *v1.Pod) *metric.Family {
 			ms := []*metric.Metric{}
-			labelKeys := []string{"container", "image_spec", "image", "image_id", "container_id"}
+			labelKeys := []string{"container", "image_spec", "image", "image_id", "container_id", "type"}
```
Comment (mrueg) [inline internal/store/pod.go:680]:
```suggestion
			labelKeys := []string{"container", "image_spec", "image", "image_id", "container_id", "restart_policy"}
```

let's use the kubernetes specific labels

Comment (changhyuni) [reply internal/store/pod.go]: nice but the pod restart policy has a label key name of type, is that okay?
https://github.com/kubernetes/kube-state-metrics/blob/main/internal/store/pod.go#L1191C11-L1191C11

Comment (mrueg) [reply internal/store/pod.go]: yes, that one is in a separate metric, I think if this is in an Info metric it's too generic to call it type.

Comment (changhyuni) [reply internal/store/pod.go]: @mrueg 
I understand. From that perspective, restart_policy would be better.

File: docs/pod-metrics.md [hunk 2]
```diff
@@ -36,7 +36,7 @@
...
+| kube_pod_init_container_info                          | Gauge       | Information about an init container in a pod                                                                                                                                        |                                                | `container`=&lt;container-name&gt; <br> `pod`=&lt;pod-name&gt; <br> `namespace`=&lt;pod-namespace&gt; <br> `image`=&lt;image-name&gt; <br> `image_id`=&lt;image-id&gt; <br> `image_spec`=&lt;image-spec&gt; <br> `container_id`=&lt;containerid&gt; <br> `uid`=&lt;pod-uid&gt; <br> `type`=&lt;Always&gt;                                                                                    | STABLE       | -      |
```
Comment (mrueg) [inline docs/pod-metrics.md:39]:
```suggestion
| kube_pod_init_container_info                          | Gauge       | Information about an init container in a pod                                                                                                                                        |                                                | `container`=&lt;container-name&gt; <br> `pod`=&lt;pod-name&gt; <br> `namespace`=&lt;pod-namespace&gt; <br> `image`=&lt;image-name&gt; <br> `image_id`=&lt;image-id&gt; <br> `image_spec`=&lt;image-spec&gt; <br> `container_id`=&lt;containerid&gt; <br> `uid`=&lt;pod-uid&gt; <br> `type`=&lt;restart-policy&gt;                                                                                    | STABLE       | -      |
```

One last change.

Comment (changhyuni) [reply docs/pod-metrics.md]: `<br> `restart_policy`=&lt;Always&gt;` right?

Comment (mrueg) [reply docs/pod-metrics.md]: The `Always` should be replaced by a placeholder

Comment (changhyuni) [reply docs/pod-metrics.md]: @mrueg 
i fixed

File: docs/pod-metrics.md [hunk 3]
```diff
@@ -36,7 +36,7 @@
...
+| kube_pod_init_container_info                          | Gauge       | Information about an init container in a pod                                                                                                                                        |                                                | `container`=&lt;container-name&gt; <br> `pod`=&lt;pod-name&gt; <br> `namespace`=&lt;pod-namespace&gt; <br> `image`=&lt;image-name&gt; <br> `image_id`=&lt;image-id&gt; <br> `image_spec`=&lt;image-spec&gt; <br> `container_id`=&lt;containerid&gt; <br> `uid`=&lt;pod-uid&gt; <br> `type`=&lt;restart-policy&gt;                                                                                   | STABLE       | -      |
```
Comment (mrueg) [inline docs/pod-metrics.md:39]:
```suggestion
| kube_pod_init_container_info                          | Gauge       | Information about an init container in a pod                                                                                                                                        |                                                | `container`=&lt;container-name&gt; <br> `pod`=&lt;pod-name&gt; <br> `namespace`=&lt;pod-namespace&gt; <br> `image`=&lt;image-name&gt; <br> `image_id`=&lt;image-id&gt; <br> `image_spec`=&lt;image-spec&gt; <br> `container_id`=&lt;containerid&gt; <br> `uid`=&lt;pod-uid&gt; <br> `restart_policy`=&lt;restart-policy&gt;                                                                                   | STABLE       | -      |
```

Comment (mrueg) [reply docs/pod-metrics.md:39]: Almost :)

**Comment by mrueg** [comment]
/lgtm 

Thanks for your contribution!

**Pull request merged**